src/frameobject.ts

```diff
@@ -141,6 +143,36 @@ class Py_FrameObject implements IPy_FrameObj {
             }
         }
     }
+
+    emptyStack() {
+        while (this.stack.length > 0) {
+            this.pop();
+        }
+    }
+    raise_exception_here(t: Thread, message: string, type: string): void {
+        var val: IPy_Object = (<any> builtins)[type];
+        t.exc = val;
+        this.push(val);
+        t.addToTraceback(message);
+        this.tryCatchException(t);
+    }
+
+    tryCatchException(t: Thread): boolean {
+        // Whenever an exception occurs, tries to find a handler and if it can't outputs the traceback 
+        // search in current frame
+        if (!find_exception_handler(this)) {
+            // Search for exception handler in previous frames
```

Comment:
As a side note, I know that the frame technically has a back pointer to its caller -- but we shouldn't use it to crawl up the stack. There are certain cases where that back pointer might not be available, such as module loading or calls from native functions. I haven't experimented with CPython yet to see if it's ever unavailable, but I bet there are corner cases.